packages/compass-e2e-tests/helpers/commands/run-find-operation.ts

```diff
@@ -102,14 +105,14 @@ async function maybeResetQuery(browser: CompassBrowser, tabName: string) {
     // look up the current resultId
     const initialResultId = await browser.getQueryId(tabName);
 
-    await resetButton.click();
-
-    // wait for the button to become disabled which should happen once it reset
-    // all the filter fields
     await browser.waitUntil(async () => {
+      // In some very rare cases on particularly slow machines in CI (looking at
+      // you macos hosts) clicking doesn't register on the first try, to work
+      // around that, we try to click with pause until the button is disabled
+      await browser.clickVisible(Selectors.queryBarResetFilterButton(tabName));
+      await browser.pause(50);
```

Comment:
This was the reason for flake in tests, I think more than anything switching to using our `clickVisible` helper instead of `el.click` is what fixes it, but just in case I am keeping the logic that is clicking in a loop until we make sure that reset button was clicked